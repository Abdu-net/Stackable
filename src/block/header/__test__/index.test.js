import { edit, save } from '../index'

describe( 'Header', () => {
	test( 'block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				title: 'Heading Title',
				subtitle: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus congue tincidunt nisit ut pretium. Duis blandit, tortor et suscipit tincidunt, dolor metus mattis neque, ac varius magna nibh ac tortor.',
				url: 'https://images.unsplash.com/photo-1506269351850-0428eaed2193?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=a1512d7659e4817df8217cdb9aa09d7a&auto=format&fit=crop&w=1050&q=80',
				buttonURL: 'http://www.gambit.ph/',
				titleColor: '#ffffff',
				subtitleColor: '#ffffff',
				buttonText: 'Button',
				buttonColor: '#2091e1',
				buttonTextColor: '#ffffff',
				size: 'normal',
				cornerButtonRadius: 4,
				contentAlign: 'center',
				id: '',
				backgroundColor: '#000000',
				opacity: 5,
			},
		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				title: 'Heading Title',
				subtitle: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus congue tincidunt nisit ut pretium. Duis blandit, tortor et suscipit tincidunt, dolor metus mattis neque, ac varius magna nibh ac tortor.',
				url: 'https://images.unsplash.com/photo-1506269351850-0428eaed2193?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=a1512d7659e4817df8217cdb9aa09d7a&auto=format&fit=crop&w=1050&q=80',
				buttonURL: 'http://www.gambit.ph/',
				titleColor: '#ffffff',
				subtitleColor: '#ffffff',
				buttonText: 'Button',
				buttonColor: '#2091e1',
				buttonTextColor: '#ffffff',
				size: 'normal',
				cornerButtonRadius: 4,
				contentAlign: 'center',
				id: '',
				backgroundColor: '#000000',
				opacity: 5,
			},
		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
