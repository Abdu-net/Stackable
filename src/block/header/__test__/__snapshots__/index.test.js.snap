// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header block saved block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-header ugb-header ugb-header--v2 ugb--background-opacity-2 ugb-header--design-design ugb--has-background ugb--has-background-image ugb--shadow-1 className"
    style={
      Object {
        "--ugb-background-color": "#123456",
        "backgroundAttachment": "fixed",
        "backgroundColor": "#123456",
        "backgroundImage": "url(backgroundImageURL)",
        "borderRadius": 6,
        "textAlign": "contentAlign",
      }
    }
  >
    <div
      className="ugb-content-wrapper"
    >
      <Unknown
        className="ugb-header__title"
        format="string"
        style={
          Object {
            "color": "#123456",
          }
        }
        tagName="h2"
        value="title"
      />
      <Unknown
        className="ugb-header__subtitle"
        format="string"
        style={
          Object {
            "color": "#123456",
          }
        }
        tagName="p"
        value="subtitle"
      />
      <Unknown
        align="contentAlign"
        backgroundColor="#123456"
        borderRadius={2}
        buttonNewTab={true}
        color="#123456"
        design="buttonDesign"
        icon="buttonIcon"
        size="size"
        text="buttonText"
        url="buttonURL"
      />
    </div>
  </div>
</BlockContentProvider>
`;

exports[`Header block saved default block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-header ugb-header ugb-header--v2 ugb--background-opacity-5 ugb-header--design-basic ugb--has-background"
    style={
      Object {
        "--ugb-background-color": undefined,
        "backgroundAttachment": undefined,
        "backgroundColor": "#000000",
        "backgroundImage": undefined,
        "borderRadius": undefined,
        "textAlign": "center",
      }
    }
  >
    <div
      className="ugb-content-wrapper"
    >
      <Unknown
        className="ugb-header__title"
        format="string"
        style={
          Object {
            "color": "#ffffff",
          }
        }
        tagName="h2"
        value="Title for This Block"
      />
      <Unknown
        className="ugb-header__subtitle"
        format="string"
        style={
          Object {
            "color": "#ffffff",
          }
        }
        tagName="p"
        value="Description for this block. Use this space for describing your block. Any text will do."
      />
      <Unknown
        align="center"
        borderRadius={4}
        buttonNewTab={false}
        color="#ffffff"
        design="basic"
        size="normal"
        text="Button text"
        url=""
      />
    </div>
  </div>
</BlockContentProvider>
`;
